blog/2024-02-27-extraction-matters-most/index.mdx

```diff
@@ -0,0 +1,135 @@
+We ran the experiment and evaluation on the cross-product of all of these factors.
+
+The results are available in a [zip file][results_zip] and the [analysis notebook][notebook] used to produce the charts is available in Google Colab.
+For each of the choices we show a heatmap comparing the median of the relative differences Context F1 between the choices.
+
+### Extraction
+
+Our suspicion was that extraction would play a significant role in the results.
+After all, if the wrong text is extracted from a PDF, or it is extracted in a confusing manner there isn’t much that can be done to retrieve the right content.
+
+The results largely supported this.
+In this visualization you can see that switching from `pymupdf` to `unstructured` produces an 87% increase in f1 score, and changing from `pypdf` to `unstructured` produces a 72% increase.
+
+![Heatmap of Extraction](./heatmap_extractor.svg)
+
+### Chunking & Indexing
+
+On the indexing front, the results were somewhat surprising.
+We expected more sophisticated chunking strategies like `semantic` to outperform the basic `recursive` chunking.
+But the results suggest that `recursive` generally outperforms the other techniques.
+As you can see, `recursive` indexing produces context f1 scores 81% better than `semantic`, and 56% better than parent indexing.
+
+Generating the questions-answered is a relatively expensive process (applying an LLM during indexing) and didn’t seem to significantly improve upon the baseline chunking.
+
+![Heatmap of Indexing strategies](./heatmap_indexing.svg)
+
+In the future, we’d like to rerun these experiments on larger datasets to see if these advanced indexing techniques become more important in different cases.
+Until then, this suggests that for simple RAG use cases, simple indexing strategies like `recursive` work well.
+
+### Search
+
+Many vector store retrievers support basic similarity search as well as a “maximum marginal relevance” or MMR.
+This is a technique to retrieve more diverse chunks with the hopes of producing richer contexts.
+We expected MMR to outperform similarity, but in our experiments we saw the opposite - MMR reduced f1 scores by 13%.
+
+![Heatmap of search strategies](./heatmap_search.svg)
+
+Our hypothesis is that because the dataset has a single PDF, there isn’t much benefit to increased diversity.
+This is another experiment we’d like to re-run on different datasets in the future.
+
+### Embeddings
+
+On embeddings, we expected the larger (and better) embeddings from OpenAI ada-002 with 1536 dimensions to outperform [bge-small-en-v1.5](https://huggingface.co/BAAI/bge-small-en-v1.5) which only has 512 dimensions.
+While we did see some improvement (2%) it was relatively minor.
```

Comment:
I think the `NaN` in the results are causing these numbers to be weird -- I'm seeing no difference between embeddings. I need to dig into that further before considering these finalized.